x-pack/plugins/stack_alerts/public/rule_types/es_query/rule_common_expressions/rule_common_expressions.tsx

```diff
@@ -67,18 +83,54 @@ export const RuleCommonExpressions: React.FC<RuleCommonExpressionsProps> = ({
   excludeHitsFromPreviousRun,
   onChangeExcludeHitsFromPreviousRun,
 }) => {
+  const [isExcludeHitsDisabled, setIsExcludeHitsDisabled] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (groupBy) {
+      setIsExcludeHitsDisabled(groupBy !== builtInGroupByTypes.all.value);
+    }
+  }, [groupBy]);
   return (
     <>
       <EuiTitle size="xs">
         <h4>
           <FormattedMessage
             id="xpack.stackAlerts.esQuery.ui.conditionsPrompt"
-            defaultMessage="Set the threshold and time window"
+            defaultMessage="Set the group, threshold and time window"
```

Comment:
Our [writing style guide](https://brand.elastic.co/302f66895/p/194a3b-writing-style-guide) recommends using serial commas:
```suggestion
            defaultMessage="Set the group, threshold, and time window"
```